contrib/infiniband/ibvctx.c

```diff
@@ -1116,16 +1116,26 @@ int _get_cq_event(struct ibv_comp_channel * channel,
   flags = fcntl(internal_channel->real_channel->fd, F_GETFL, NULL);
 
   // We need to change the call to non-blocking mode
-  if (flags & O_NONBLOCK == 0) {
+  // FIXME: we need to take care of the case where fd is already
+  // changed to non-blocking mode by the user.
```

Comment:
Right. But it's hard to tell when the user decides to change the mode. If we can assume the user does it before any call to the function, then I can add a static variable to indicate the mode before the first call to the function.